cpu/stm32/periph/spi.c

```diff
@@ -162,7 +183,12 @@ int spi_acquire(spi_t bus, spi_cs_t cs, spi_mode_t mode, spi_clk_t clk)
     /* enable SPI device clock */
     periph_clk_en(spi_config[bus].apbbus, spi_config[bus].rccmask);
     /* enable device */
-    uint8_t br = spi_divtable[spi_config[bus].apbbus][clk];
+    if (clk != clocks[bus]) {
+        dividers[bus] = _get_clkdiv(&spi_config[bus], clk);
+        clocks[bus] = clk;
+    }
+    uint8_t br = dividers[bus];
```

Comment:
You could go one step further and cache `cr1_settings`.

Reply:
See https://github.com/RIOT-OS/RIOT/pull/14749#discussion_r470596815, that only works if both the SPI mode and the HW chip-select settings are also included in the cache. Could be a good follow-up, but I don't want to include it here.